lib/AST/ASTPrinter.cpp

```diff
@@ -64,22 +65,7 @@ void PrintOptions::clearSynthesizedExtension() {
   TransformContext.reset();
 }
 
-static bool isPublicOrUsableFromInline(const ValueDecl *VD) {
-  AccessScope scope =
-      VD->getFormalAccessScope(/*useDC*/nullptr,
-                               /*treatUsableFromInlineAsPublic*/true);
-  return scope.isPublic();
-}
-
-static bool contributesToParentTypeStorage(const AbstractStorageDecl *ASD) {
-  auto *DC = ASD->getDeclContext()->getAsDecl();
-  if (!DC) return false;
-  auto *ND = dyn_cast<NominalTypeDecl>(DC);
-  if (!ND) return false;
-  return !ND->isResilient() && ASD->hasStorage() && !ASD->isStatic();
-}
-
-PrintOptions PrintOptions::printParseableInterfaceFile() {
+PrintOptions PrintOptions::printParseableInterfaceFile(ModuleDecl *module) {
```

Comment:
I'm honestly starting to think this doesn't belong on PrintOptions at all. Only the reusable or generic ones really deserve factory constructors; the rest should live at their use sites. That doesn't have to change in this PR, but it kind of lines up with the spiltting out to a separate file.